[WWST-1697]Updated the inappropriate lock value for ONEAPP-9767

vangoran opened the pull request.

vangoran committed:
[WWST-1697]Updatedinappropriate lock value ONEAPP-9767

vangoran commented:
@greens @tpmanley @dkirker 
According to ONEAPP-9767, I updated inappropriate lock value and tiles for Samsung Smart Doorlock, please review it.

tpmanley commented on devicetypes/samsungsds/samsung-smart-doorlock.src/samsung-smart-doorlock.groovy:
```diff
@@ -366,7 +364,7 @@ private def parseCommandResponse(String description) {
 				break
 			case 7: // Door opened for over 30 seconds
-				responseMap = [ name: "lock", value: "autolockError", descriptionText: "Door remains opened for over 30 seconds" ]
+				responseMap = [ name: "tamper", value: "detected", descriptionText: "Door remains opened for over 30 seconds", isStateChange: true ]
```
There doesn't seem to be anything that will clear `tamper` once set. Is that what you intended?

tpmanley replied:
It's fine if this isn't addressed in this PR since it's not a new issue but I do recommend that you figure out a way to clear the tamper, either after a delay or when the user takes a specific action

FenMei007S commented:
@tpmanley, i forward a email to u to explain the modification

tpmanley commented on devicetypes/samsungsds/samsung-smart-doorlock.src/samsung-smart-doorlock.groovy:
```diff
@@ -37,8 +37,6 @@ metadata {
 				attributeState "locked", label:'locked', action:"lock.unlock", icon:"st.locks.lock.locked", backgroundColor:"#00A0DC", nextState:"unlocking"
 				attributeState "unlocked", label:'unlocked', icon:"st.locks.lock.unlocked", backgroundColor:"#ffffff"
 				attributeState "unknown", label:"unknown", icon:"st.locks.lock.unknown", backgroundColor:"#ffffff"
-				attributeState "unlocking", label:'unlocking', icon:"st.locks.lock.unlocked", backgroundColor:"#ffffff"
```
I don't think you need/want to remove this state since it's still used by `nextState` in the "locked" state

vangoran commented:
@tpmanley
I made a change for your addressed comment, every alarm code is commented out per our current scope for deployment. And, from the tile section, nextState of unlocking is removed as it seems unnecessary.

tpmanley commented:
This change was already merged which addresses the incorrect lock value: https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/4123. Do you still want to make additional changes to the DTH? I'm not opposed to it since I do think something should be done about the `tamper` (either adding something to clear it or not setting in the first place).

greens commented on devicetypes/samsungsds/samsung-smart-doorlock.src/samsung-smart-doorlock.groovy:
```diff
@@ -351,31 +349,31 @@ private def parseCommandResponse(String description) {
 		def alarmCode = Integer.parseInt(data[0], 16)
 		switch (alarmCode) {
 			case 0: // Deadbolt Jammed
```
wouldn't a block commenting out this whole method be better here?

vangoran replied:
Good suggestion, I comment out this whole block

vangoran commented:
> This change was already merged which addresses the incorrect lock value: #4123. Do you still want to make additional changes to the DTH? I'm not opposed to it since I do think something should be done about the `tamper` (either adding something to clear it or not setting in the first place).

Besides this change: #4123, inappropriate attributeStates are removed, so I'd like to commit this new one.

vangoran committed:
Merge branch 'master' into WWST-1697

vangoran commented:
Could anybody help to merge this PR?

greens commented:
Merged
